graphrag/query/factories.py

```diff
@@ -114,7 +115,7 @@ def get_local_search_engine(
     """Create a local search engine based on data + configuration."""
     llm = get_llm(config)
     text_embedder = get_text_embedder(config)
-    token_encoder = tiktoken.get_encoding(config.encoding_model)
+    token_encoder = tiktoken.encoding_for_model(config.llm.model)
```

Comment:
Should we do this automatic lookup of encoding model via the LLM model name or rely on the user to provide the correct value?

This will error out for anyone not using OpenAI models.

Perhaps we compromise here and use the user-provided encoding model name as before. However we add a check to see if the LLM model name exists in the tiktoken library and if not, log a warning message to the user that the encoding model name is not recognized and we will fallback to the user-provided ending model name.

if the encoding model name IS in tiktoken and does not match the user-provided value, then should we error out or just log a warning?